### ISSUE TITLE
/v1/module handles fields parameter in a way that produces single-value arrays

### ISSUE THREAD

**Issue by tsibley** [metacpan/metacpan-api#580]
Compare:

```
$ curl -fsS https://api.metacpan.org/v0/module/Pod::Cpandoc?fields=distribution
{
   "distribution" : "Pod-Cpandoc"
}

$ curl -fsS https://fastapi.metacpan.org/v1/module/Pod::Cpandoc?fields=distribution
{
   "distribution" : [
      "Pod-Cpandoc"
   ]
}

$ curl -fsS https://fastapi.metacpan.org/v1/module/Pod::Cpandoc
{
   …,
   "distribution" : "Pod-Cpandoc",
   …
}
```

I'll take a look at this in the next couple days if no one else does.

**Comment by ranguard** [comment]
I know the 'utility' of flattening arrays is nice, but if it leads to our code having lots of these we should seriously consider just keeping them as arrays all  the time

**Comment by ranguard** [comment]
s/keeping them/keeping results/

**Comment by tsibley** [comment]
Well, it's a huge PITA for consumers which otherwise would just be able to change endpoints and be done with it.

**Comment by tsibley** [comment]
For this particular endpoint, we obviously take pains to collapse single-valued arrays, but miss the case when the `fields` parameter is specified or something.  (I haven't looked at the code yet.)

**Comment by tsibley** [comment]
We can also refactor the controller/model code such that there's only a few places where the output gets filtered through our munger.

**Comment by tsibley** [comment]
Fixed in #581.
